tracing-core/src/metric.rs

```diff
@@ -0,0 +1,128 @@
+//! Metrics represent single points in time during the execution of a program.
+use crate::parent::Parent;
+use crate::span::Id;
+use crate::{field, Metadata};
+
+/// `Metric`s represent single points in time where something occurred during the
+/// execution of a program.
+///
+/// An `Metric` can be compared to a log record in unstructured logging, but with
+/// two key differences:
+/// - `Metric`s exist _within the context of a [span]_. Unlike log lines, they
+///   may be located within the trace tree, allowing visibility into the
+///   _temporal_ context in which the metric occurred, as well as the source
+///   code location.
+/// - Like spans, `Metric`s have structured key-value data known as _[fields]_,
+///   which may include textual message. In general, a majority of the data
+///   associated with an metric should be in the metric's fields rather than in
+///   the textual message, as the fields are more structured.
+///
+/// [span]: super::span
+/// [fields]: super::field
+#[derive(Debug)]
+pub struct Metric<'a> {
+    fields: &'a field::ValueSet<'a>,
+    metadata: &'static Metadata<'static>,
+    parent: Parent,
```

Comment:
This would have to be changed of course. Like I mention in the body of my PR, the usual data we gather for Events might not all be relevant for metrics.